lib/internal/modules/package_json_reader.js

```diff
@@ -104,29 +117,56 @@ function readPackage(requestPath) {
 /**
  * Get the nearest parent package.json file from a given path.
  * Return the package.json data and the path to the package.json file, or undefined.
- * @param {string} checkPath The path to start searching from.
- * @returns {undefined | {data: import('typings/internalBinding/modules').PackageConfig, path: string}}
+ * @param {URL['href'] | URL['pathname']} startPath The path to start searching from.
+ * @param {boolean} everything Whether to include the full contents of the package.json.
+ * @returns {undefined | {
+ *   data: everything extends true ? FullPackageConfig : RecognizedPackageConfig,
+ *   path: URL['pathname'],
+ * }}
  */
-function getNearestParentPackageJSON(checkPath) {
-  const result = modulesBinding.getNearestParentPackageJSON(checkPath);
+function getNearestParentPackageJSON(startPath, everything = false) {
+  if (typeof startPath !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('startPath', 'string', startPath);
+  }
+  if (
+    everything !== undefined &&
+    typeof everything !== 'boolean'
+  ) {
+    throw new ERR_INVALID_ARG_TYPE('everything', 'boolean', everything);
+  }
+
+  if (everything) {
+    const result = modulesBinding.getNearestRawParentPackageJSON(startPath);
+
+    return {
+      data: {
+        __proto__: null,
```

Comment:
Having null-prototype object returned to the object feels weird

Reply:
I agree. I did it for consistency between `everything` (not `everything` is used internally and needs the null prototype).

Reply:
That feels like another reason not to expose `everything` and let the user parse the JSON themselves. We should be freezing it to avoid folks mutating it – but really, why do we bother? The API would be much simpler if it only returns a path

Reply:
The reason is because we've already done the work, and handled the caching. Doing it this way also provides better performance. Users having to do exactly the same thing is the whole reason we as engineers create utilities, and there is basically no chance any user would not subsequently parse it.

Freezing seems like a good idea.